### PR TITLE
end to end multilabel (region) support and V2's of the Pipeline, along with some presets and functions support them.

### PR DESCRIPTION
**Pull request by Sllambias**
Version 2.0.0.
Reason: PR is not backwards compatible.

Adds:

- [x] Functional evaluation (evaluate_folder_segm, evaluate_multilabel_folder_segm, etc.)
- [x] ManagerV2 with better defaults
- [x] PlannerV2 with better defaults
- [x] remove some noise from plans
- [x] Support for multilabel (e.g. one hot label regions) saving and inference and evaluation
- [x] Generic (medical) augmentation presets
- [x] Option to change oversampling frequency
- [x] UNetS preset (as function)

### CODE REVIEW

File: yucca/functional/evaluation/evaluate_folder.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+from yucca.functional.utils.nib_utils import get_nib_spacing
+from yucca.functional.evaluation.obj_metrics import get_obj_stats_for_label
+from yucca.functional.evaluation.surface_metrics import get_surface_metrics_for_label
+from tqdm import tqdm
+from batchgenerators.utilities.file_and_folder_operations import join
+from sklearn.metrics import confusion_matrix
+from yucca.functional.evaluation.metrics import auroc
+
+
+def evaluate_multilabel_folder_segm(
```
Comment (asbjrnmunk) [inline yucca/functional/evaluation/evaluate_folder.py:16]: I think this is so cool to have. I think it would be nice to spend just a few minutes rewriting this code, so it is a bit less duplicated and I think those functions might also be useful in Agave and other projects.

File: yucca/functional/transforms/label_transforms.py [hunk 2]
```diff
@@ -13,36 +13,59 @@ def translate_region_labels(regions: dict, labels: dict) -> dict:
...
         region_labels = regions[region]["labels"]
+        assert isinstance(region_labels[0], int)
         region_canvas[:, channel] = np.isin(label[:, 0], region_labels)
     region_canvas = region_canvas.astype(np.uint8)
 
     return region_canvas
 
 
-# Currently not used, and incorrect with new regions format.
+def convert_labels_to_regions(data, regions):
```
Comment (asbjrnmunk) [inline yucca/functional/transforms/label_transforms.py:30]: Maybe i am not reading this correct, but shouldn't this use the new "priority" property in the region dict?

**Review by asbjrnmunk — APPROVED** [review]
Yucca 2.0 🚀

File: yucca/functional/transforms/label_transforms.py (hunk 2 above)
Comment (Sllambias) [reply yucca/functional/transforms/label_transforms.py:30]: Not quite, priority is something we should look to use when we go the other way around (where it should be added) so we inflate label of shape (1, h, w, d) to (r, h, w, d) in the order of the regions, then when we go back to (1, h, w, d) we want to check that we put the "overlay" the regions in the correct order, so the priority 1 region overwrites any other labels. That however, is not yet implemented which it should be - but it is not necessary for what we do atm, since the brats eval never goes back from regions to labels
